docs/base/rust_plugin.rst

```diff
@@ -0,0 +1,140 @@
+This option may significantly impact the build performance but
+reducing the final binary size and improve the runtime performance.
+This will forcibly enable LTO for all the crates you specified,
+regardless of whether the projects have the LTO option enabled
+in the Cargo.toml file.
+
+This is equivalent to the ``lto = "fat"`` option in the Cargo.toml file.
+
+If you want maximum runtime performance as possible,
+see the :ref:`Performance tuning<perf-tuning>` section below.
+
+Environment variables
+---------------------
+
+This plugin also sets the PATH environment variable in the build environment
+so that the Rust compiler can be found.
+
+Some environment variables may also influence the Rust compiler or Cargo build tool.
+For more information, see `Cargo documentation <https://doc.rust-lang.org/cargo/reference/environment-variables.html>`_ for the details.
+
+Dependencies
+------------
+
+By default this plugin uses Rust toolchain binaries from the Rust upstream.
+If this is not desired, you can set ``rust-deps: ["rustc", "cargo"]`` and
+``rust-channel: "none"`` in the part definition to override the default behaviour.
+
+.. _perf-tuning:
+Performance tuning
+-------------------
+
+Many Rust programs boast their performance over similar programs implemented in other
+programming languages.
+To get even better performance, you might want to follow the tips below.
+Keep in mind that due to individual differences between different projects, some of the
+optimisations may not work as expected or even incur performance penalties.
+
+* Use the :ref:`rust-use-global-lto` option to enable LTO support. This is suitable for most
+  projects. However, analyzing the whole program during the build time requires more memory and CPU time.
+
+* Specify ``codegen-units=1`` in ``Cargo.toml`` to reduce LLVM parallelism. This may sound counter-intuitive,
+  but reducing code generator threads could improve the quality of generated machine code.
+  This option will also reduce the build time performance since the code generator uses only one thread per translation unit.
+
+* Disable ``incremental=true`` in ``Cargo.toml`` to improve inter-procedural optimisations. Many projects may have
+  already done this for the release profile. You should check if that is the case for your project.
+
+* (Advanced) Perform cross-language LTO. This requires installing the correct version of LLVM/Clang and setting the right environment variables.
+  You must know which LLVM version of your selected Rust toolchain is using.
+  For example, Rust 1.71 uses LLVM 16 because you can see it bundles a ``libLLVM-16-rust-1.71.1-stable.so`` file in the ``lib`` directory.
+  In this case, you want to install ``clang-16`` and ``lld-16`` from the Ubuntu archive.
+
+  * You will need to set these environment variables for Clang:
+      .. code-block:: bash
+
+        export CC=clang-16
+        export CXX=clang++-16
+        export CFLAGS=-flto=fat
+        export CXXFLAGS=-flto=fat
```

Comment:
This might be nice to provide an example `parts` YAML snippet for rather than the bash. E.g.:

```suggestion
      .. code-block:: yaml

        parts:
          my-app:
            plugin: rust
            source: .
            build-packages:
              - clang-16
              - lld-16
            build-environment:
              - CC: clang-16
              - CXX: clang++-16
              - CFLAGS: -flto=fat
              - CXXFLAGS: -flto=fat
```